docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2001 @@
+<!---
+NOTE: The OpenAPI reference (../../rpc) is injected into the documentation during
+the CometBFT docs build process. See https://github.com/cometbft/cometbft-docs/
+for details.
+--->
+
+You can also read the [RPC specification](../../spec/rpc) for more information.
+
+### rpc.cors_allowed_origins
+A list of origins a cross-domain request can be executed from.
+```toml
+cors_allowed_origins = []
+```
+
+| Value type          | array of string                            |                      |
+|:--------------------|:-------------------------------------------|----------------------|
+| **Possible values** | `[]`                                       | disable CORS support |
+|                     | `["*"]`                                    | allow any origin     |
+|                     | array of strings containing domain origins |                      |
+
+Domain origins are fully qualified domain names with protocol prefixed, for example `"https://cometbft.com"` or
+they can contain exactly one wildcard to extend to multiple subdomains, for example: `"https://*.myapis.com"`.
+
+Example:
+
+Allow only some subdomains for CORS requests:
+```toml
+cors_allowed_origins = ["https://www.cometbft.com", "https://*.apis.cometbft.com"]
+```
+
+### rpc.cors_allowed_methods
+A list of methods the client is allowed to use with cross-domain requests.
+```toml
+cors_allowed_methods = ["HEAD", "GET", "POST", ]
+```
+
+| Value type                              | array of string |
+|:----------------------------------------|:----------------|
+| **Possible string values in the array** | `"HEAD"`        |
+|                                         | `"GET"`         |
+|                                         | `"POST"`        |
+
+You can read more about the methods in the
+[Mozilla CORS documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).
+
+### rpc.cors_allowed_headers
+A list of headers the client is allowed to use with cross-domain requests.
+```toml
+cors_allowed_headers = ["Origin", "Accept", "Content-Type", "X-Requested-With", "X-Server-Time", ]
+```
+
+| Value type                              | array of string      |
+|:----------------------------------------|:---------------------|
+| **Possible string values in the array** | `"Accept"`           |
+|                                         | `"Accept-Language"`  |
+|                                         | `"Content-Language"` |
+|                                         | `"Content-Type"`     |
+|                                         | `"Range"`            |
+
+The list of possible values are from the [Fetch spec](https://fetch.spec.whatwg.org/#cors-safelisted-request-header)
+which defines `Origin` as a forbidden value. Read the
+[Mozilla CORS documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) and do your own tests, if you want
+to use this key.
+
+<!--- Possibly, we should clarify the allowed values better. --->
```

Comment:
we can improve in the future if needed or there are requests from users for it. Since CORS is a widely adopted concept, I believe it's out of scope to provide a very detailed information here.